apps/shopify/src/skuResolvers.js

```diff
@@ -177,6 +177,24 @@ export const makeCollectionSearchResolver = async (sdk) => {
   return (search) => pagination.fetchNext(search);
 };
 
+export const filterAndDecodeValidIds = (skus, skuType) => {
+  const validIds = skus
+    .map((sku) => {
+      try {
+        // If not valid base64 window.atob will throw
+        const decodedId = atob(sku);
+        return { decodedId };
+      } catch (error) {
+        return null;
+      }
+    })
+    .filter((sku) => sku && new RegExp(`^gid.*${skuType}`).test(sku.decodedId))
+    /** changed to unencodedId  as API format changed and in future support for base64 will be removed so 
+        passing the gid based format to the API for fetching the productVariant for particular product */
+    .map(({ decodedId }) => decodedId);
```

Comment:
Why are we mapping to an object and unwrapping it here? Let's directly map to the decoded id